server/src/main/java/org/elasticsearch/index/mapper/FieldTypeLookup.java

```diff
@@ -132,6 +132,10 @@ MappedFieldType get(String field) {
         if (fullNameToFieldType.isEmpty()) {
             return Set.of();
         }
+        if (dynamicKeyLookup.get(field) != null) {
+            return Set.of(field);
+        }
```

Comment:
I'm not entirely sure if this gets in the way of other logic, but in the case of a flattened subfield we would return the root path otherwise without this extra step before the rest of the logic here. Happy to hear if there are better places to but this overwrite.

Reply:
I think this is correct, it's the only way to distinguish between a flattened field and a multifield.